src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -131,59 +151,68 @@ public object VisitSqlRoot(SqlRootExpression expression, SearchOptions context)
                 StringBuilder.AppendLine();
             }
 
-            StringBuilder.Append("FROM ").Append(VLatest.Resource).Append(" ").Append(resourceTableAlias);
-
-            if (expression.TableExpressions.Count == 0 &&
-                !_isHistorySearch &&
-                expression.DenormalizedExpressions.Any(e => e.AcceptVisitor(ExpressionContainsParameterVisitor.Instance, SearchParameterNames.ResourceType)))
-            {
-                // If this is a simple search over a resource type (like GET /Observation)
-                // make sure the optimizer does not decide to do a scan on the clustered index, since we have an index specifically for this common case
-                StringBuilder.Append(" WITH(INDEX(").Append(VLatest.Resource.IX_Resource_ResourceTypeId_ResourceSurrgateId).AppendLine("))");
-            }
-            else
-            {
-                StringBuilder.AppendLine();
-            }
-
-            if (expression.TableExpressions.Count > 0)
+            if (selectingFromResourceTable)
             {
-                StringBuilder.AppendLine().Append("INNER JOIN ").AppendLine(TableExpressionName(_tableExpressionCounter));
-                StringBuilder.Append("ON ").Append(VLatest.Resource.ResourceSurrogateId, resourceTableAlias).Append(" = ").Append(TableExpressionName(_tableExpressionCounter)).AppendLine(".Sid1");
-            }
+                StringBuilder.Append("FROM ").Append(VLatest.Resource).Append(" ").Append(resourceTableAlias);
 
-            using (var delimitedClause = StringBuilder.BeginDelimitedWhereClause())
-            {
-                foreach (var denormalizedPredicate in expression.DenormalizedExpressions)
+                if (expression.TableExpressions.Count == 0 &&
+                    !_isHistorySearch &&
+                    expression.DenormalizedExpressions.Any(e => e.AcceptVisitor(ExpressionContainsParameterVisitor.Instance, SearchParameterNames.ResourceType)) &&
+                    !expression.DenormalizedExpressions.Any(e => e.AcceptVisitor(ExpressionContainsParameterVisitor.Instance, SearchParameterNames.Id)))
```

Comment:
This is not the index to use for this kind of query:
`GET https://localhost:44348/Observation?_id=a937ba06-da1e-4f37-bbb4-848d60f26aed` because `ResourceId` is not part of this index.